File: Python/hamming.py
def valid_strand(strand):
    """Validates the strand, making sure it contains only the ACGT letters."""
    return len([letter for letter in strand if letter in "ACGT"]) == len(strand)

def distance(strand_a, strand_b):
    """
    Calculate the Hamming Distance between two DNA strands.
    Example:
    GAGCCTACTAACGGGAT
    CATCGTAATGACGGCCT
    ^ ^ ^  ^ ^    ^^    
    They have 7 differences, and therefore, the Hamming Distance is 7.
    """
    if len(strand_a) != len(strand_b):
        raise ValueError("Different lengths strands")
    if not(valid_strand(strand_a)) or not(valid_strand(strand_b)):
        return False    
    dist = 0    
    for index in range(len(strand_a)):
        if strand_a[index] != strand_b[index]:
            dist += 1 
    return dist            

File: Go/hamming.go
// Package hamming provides a way to compute the hamming
// distance between two strings.
package hamming

import (
	"errors"
)

// Distance calculates the hamming distance between two strings.
func Distance(a, b string) (int, error) {
	runeA := []rune(a)
	runeB := []rune(b)
	if len(runeA) != len(runeB) {
		return 0, errors.New("strings have different lengths")
	}
	dist := 0
	for index, element := range runeA {
		if element != runeB[index] {
			dist++
		}
	}
	return dist, nil
}
